lib/cpp/src/thrift/server/TThreadedServer.h

```diff
@@ -20,120 +20,86 @@
-                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory))
+    : TServer(processorFactory, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(new concurrency::PlatformThreadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  const boost::shared_ptr<ThreadFactory>& threadFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
-  virtual ~TThreadedServer();
+  virtual ~TThreadedServer() {};
 
   virtual void serve();
 
-  void stop() {
+  virtual void stop() {
     stop_ = true;
     serverTransport_->interrupt();
   }
 
 protected:
-  void init();
+  virtual void disposeClient(TConnectedClient *pClient);
 
-  boost::shared_ptr<ThreadFactory> threadFactory_;
+  boost::shared_ptr<concurrency::ThreadFactory> threadFactory_;
```

Comment:
/sigh
I curse Microsoft that I have to bring this up...

I've had issues with this spelling with Visual Studio 2012.  Visual Studio 2012 has a concurrency namespace.  Not 'microsoft::concurrency', but just 'concurrency'.  Please put this back to just 'ThreadFactory', or qualify it further with an 'apache::thrift::'.